service/worker/service.go

```diff
@@ -245,14 +266,14 @@ func (svc *Service) lead(ctx context.Context) error {
 
 	// create nodeCache
 	logger.Ctx(ctx).Infof("getting active nodes for cache")
-	svc.nodeCache, err = svc.nodeCore.ListKeys(svc.ctx)
+	svc.nodeCache, err = svc.nodeCore.List(svc.ctx, "")
 	if err != nil {
 		return err
 	}
 
 	// create subscription cache
 	logger.Ctx(ctx).Infof("getting active subscriptions for cache")
-	svc.subCache, err = svc.subscriptionCore.ListKeys(svc.ctx)
+	svc.subCache, err = svc.subscriptionCore.List(svc.ctx, "")
 	if err != nil {
 		return err
 	}
```

Comment:
I think there can be a race condition here, i.e a node or subs can be added or deleted between this step and when the actual watch is set. Can we update these caches as part of setting the watch itself?

Reply:
when we create watch, that gives a mandatory first callback with all data present under that key prefix, i guess we can then just remove the manual fetch and depend only on the watch, as first watch will update the empty cache